pandas/core/arrays/categorical.py

```diff
@@ -628,8 +629,11 @@ def from_codes(cls, codes, categories, ordered=False):
             categorical. If not given, the resulting categorical will be
             unordered.
         """
+        codes = np.asarray(codes)
+        if not is_integer_dtype(codes):
+            raise ValueError("codes need to be array-like integers")
```

Comment:
so can you check for float dtype here and issue a warning, something like

```
if is_float_dtype(codes):
    icodes = codes.astype('i8')
    if (icodes != codes).all():
       raise ValueError("not integer-like float codes are invalid")
    warn.warning("float codes will be disallowed in the future", FutureWarning)
    codes = icodes
```

and tests for same.